src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1758,6 +1758,28 @@ private void BindArgumentsAndNames(BracketedArgumentListSyntax argumentListOpt,
             }
         }
 
+        private bool RefMustBeObeyed(bool isDelegateCreation, ArgumentSyntax argumentSyntax)
+        {
+            if (Compilation.FeatureStrictEnabled || !isDelegateCreation)
+            {
+                return true;
+            }
+
+            switch (argumentSyntax.Expression.Kind())
+            {
+                case SyntaxKind.ParenthesizedLambdaExpression:
+                case SyntaxKind.SimpleLambdaExpression:
+                case SyntaxKind.AnonymousMethodExpression:
+                case SyntaxKind.InvocationExpression:
+                case SyntaxKind.ObjectCreationExpression:
+                case SyntaxKind.ParenthesizedExpression:
+                    // a property/indexer should also be invalid, but cannot be checked here
+                    return true;
+                default:
+                    return false;
```

Comment:
Will do. The logic is a bit reversed, but all the compat cases pass here. The ones that are rejected, are handled correctly in `BindArgumentAndName`. See the linked issue for a list I discovered that are valid.